models/fates/R/met2model.FATES.R

```diff
@@ -61,103 +67,119 @@ met2model.FATES <- function(in.path, in.prefix, outfolder, start_date, end_date,
+        if (file.exists(outfile_tem) & overwrite == FALSE) {
+          next
+        }
+        ncout_tem <- ncdf4::nc_create(outfile_tem, vars = list(var_lat,var_long,var_E,var_W,var_S,var_N), verbose = verbose)
+        put_var(ncout_tem)
+        ## surface_downwelling_longwave_flux_in_air
+        ncout_tem <- insert(ncout_tem, "FLDS", "W m-2", FLDS[tsel], dim)
         ## air_pressure
-        ncout <- insert(ncout, "PSRF", "Pa", PSRF)
-        
+        ncout_tem <- insert(ncout_tem, "PSRF", "Pa", PSRF[tsel], dim)
         ## specific_humidity
-        ncout <- insert(ncout, "SHUM", "kg/kg", SHUM)
-        
+        ncout_tem <- insert(ncout_tem, "QBOT", "kg/kg", QBOT[tsel], dim)
         ## air_temperature
-        ncout <- insert(ncout, "TBOT", "K", TBOT)
-        
+        ncout_tem <- insert(ncout_tem, "TBOT", "K", TBOT[tsel], dim)
         ## eastward_wind & northward_wind
-        ncout <- insert(ncout, "WIND", "m/s", WIND)
+        ncout_tem <- insert(ncout_tem, "WIND", "m/s", WIND[tsel], dim)
+        ncdf4::nc_close(ncout_tem)
         
-        ncdf4::nc_close(ncout)
-        
-        #   ncvar_rename(ncfile,varid="LONGXY")
-        #   ncvar_rename(ncfile,varid="LATIXY")
-        #   #     
-        #   #     double EDGEW(scalar) ;
-        #   #     EDGEW:long_name = "western edge in atmospheric data" ;
-        #   #     EDGEW:units = "degrees E" ;
-        #   EDGEW = ncvar_rename(ncfile,"EDGEW","EDGEW")
-        #   
-        #   #     double EDGEE(scalar) ;
-        #   #     EDGEE:long_name = "eastern edge in atmospheric data" ;
-        #   #     EDGEE:units = "degrees E" ;
-        #   EDGEE = ncvar_rename(ncfile,"EDGEE","EDGEE")
-        #   
-        #   #     double EDGES(scalar) ;
-        #   #     EDGES:long_name = "southern edge in atmospheric data" ;
-        #   #     EDGES:units = "degrees N" ;
-        #   EDGES = ncvar_rename(ncfile,"EDGES","EDGES") 
-        #   #     
-        #   #     double EDGEN(scalar) ;
-        #   #     EDGEN:long_name = "northern edge in atmospheric data" ;
-        #   #     EDGEN:units = "degrees N" ;
-        #   EDGEN = ncvar_rename(ncfile,"EDGEN","EDGEN")
       }
-      
-      ncdf4::nc_close(nc)
-    }  ## end file exists
-  }  ### end loop over met files
+      ncdf4::nc_close(nc) #PEcAn input file
+    }  ## end input file exists
+  }  ### end year loop over met files
   
-  PEcAn.logger::logger.info("Done with met2model.FATES")
+  #PEcAn.logger::logger.info("Done with met2model.FATES")
   
-  return(data.frame(file = paste0(outfolder, "/"), 
-                    host = c(PEcAn.remote::fqdn()), 
-                    mimetype = c("application/x-netcdf"), 
-                    formatname = c("CLM met"), 
-                    startdate = c(start_date), 
-                    enddate = c(end_date), 
-                    dbfile.name = "", 
-                    stringsAsFactors = FALSE))
+  return(LATIXY)
+ # (data.frame(file = paste0(outfolder, "/"), 
```

Comment:
this data.frame needs to be returned to the PEcAn workflow so that the convert.input function can insert the record of the new file to the DB. Please check other met2model examples to see how they return, e.g. `return(invisible(results))`